sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -476,42 +490,44 @@ def _load_all(self, **kwargs):
                 is_failover_request = True
         raise exception
 
-    def _process_configurations(self, configuration_settings: List[ConfigurationSetting]) -> Dict[str, Any]:
-        # Reset feature flag usage
-        self._uses_ai_configuration = False
-        self._uses_aicc_configuration = False
+    def _process_configurations(
+        self, configuration_settings: List[ConfigurationSetting], refreshed_feature_flags: bool
+    ) -> Dict[str, Any]:
+        if refreshed_feature_flags:
+            # Reset feature flag usage
+            self._feature_filter_usage = {}
```

Comment:
that doesn't have anything to do with processing the settings. It's relevant when flags were refreshed only

Reply:
Discussed offline to move this flag telemetry tracker reset when we determine that flags have changed.